lib/job.jsonnet

```diff
@@ -99,8 +61,8 @@ function(jobName, agentEnv={}, stepEnvFile='', patchFunc=identity) patchFunc({
     stepEnv +
     [
       { name: f, value: env[f] }
-      for f in std.objectFields(env)
-      if std.setMember(f, allowedEnvs)
+      for f in std.objectFields(agentEnv)
```

Comment:
the only different wrt to ones that were passed through here is `BULIDKITE_TIMEOUT` since no `BUILDKITE_PLUGIN*` variables were passed through, I'm not sure how much of an issue that is

Reply:
Yeah I'm thinking that should be fine to include them since we're not overwriting anything at least